consensus/istanbul/core/backlog.go

```diff
@@ -54,7 +57,19 @@ func (c *core) checkMessage(msgCode uint64, view *istanbul.View) error {
 		return errFutureMessage
 	}
 
+	// as soon as we move on to the next sequence, we discard any messages from
+	// a previous sequence as invalid old messages if they are not Commits
 	if view.Cmp(c.currentView()) < 0 {
+		// let commits from previous views through so that the handler adds them
+		// to the view's parent commits.
+		// TODO: Should we let old rounds through as well? The view.Cmp function
+		// returns <0 if the sequences are equal and uses the round as a tiebreaker.
```

Comment:
If this node will want to maximize rewards from parent seals, then I think it should keep old rounds as well.

Once a block has been "locked" by the consensus algorithm, then all of the subsequent proposals from later rounds of the same sequence should be the same (which means all those signatures should be valid and hence "reward-able").

Reply:
Actually this TODO is leftover from a previous iteration, the current functionality lets the previous sequence through, independently of its round (which indeed as you said: once a block has been locked, all later rounds are the same)